BigBaseV2/src/hooks/protections/received_clone_sync.cpp

```diff
@@ -20,21 +35,317 @@ namespace big
+						if (g->notifications.invalid_sync.log)
+							LOG(WARNING) << "Invalid vehicle model: " << "Model: " << HEX_TO_UPPER(vehicle_creation_node->m_model) << " From : " << sender->get_name();
+						if (g->notifications.invalid_sync.notify)
+							g_notification_service->push_warning(fmt::format("Invalid vehicle model from {}", sender->get_name()), fmt::format("Model: 0x{:x}", vehicle_creation_node->m_model));
+						return true;
+					}
+					break;
+				}
+				case "?AVCObjectCreationDataNode@@"_fnv1a: //CObjectCreationDataNode
+				{
+					CObjectCreationDataNode* creation_node = dynamic_cast<CObjectCreationDataNode*>(node);
+					if (is_model_a_crash_model(creation_node->m_model)) {
+						if (g->notifications.invalid_sync.log)
+							LOG(WARNING) << "Invalid object model: " << "Model: " << HEX_TO_UPPER(creation_node->m_model) << " From : " << sender->get_name();
+						if (g->notifications.invalid_sync.notify)
+							g_notification_service->push_warning(fmt::format("Invalid object model from {}", sender->get_name()), fmt::format("Model: 0x{:x}", creation_node->m_model));
+						return true;
+					}
+					else if (is_model_a_cage_model(creation_node->m_model)) {
+						if (g->notifications.invalid_sync.log)
+							LOG(WARNING) << "Cage model: " << "Model: " << HEX_TO_UPPER(creation_node->m_model) << " From : " << sender->get_name();
+						if (g->notifications.invalid_sync.notify)
+							g_notification_service->push_warning(fmt::format("Cage model from {}", sender->get_name()), fmt::format("Model: 0x{:x}", creation_node->m_model));
+						return true;
+					}
+					break;
+				}
+				case "?AVCPlayerAppearanceDataNode@@"_fnv1a: //CPlayerAppearanceDataNode
+				{
+					CPlayerAppearanceDataNode* player_appearance_node = dynamic_cast<CPlayerAppearanceDataNode*>(node);
+					if (is_model_a_crash_model(player_appearance_node->m_model_hash)) {
+						if (g->notifications.invalid_sync.log)
+							LOG(WARNING) << "Invalid player model: " << "Model: " << HEX_TO_UPPER(player_appearance_node->m_model_hash) << " From : " << sender->get_name();
+						if (g->notifications.invalid_sync.notify)
+							g_notification_service->push_warning(fmt::format("Invalid player model from {}", sender->get_name()), fmt::format("Model: 0x{:x}", player_appearance_node->m_model_hash));
+						return true;
+					}
+					break;
+				}
+				case "?AVCSectorDataNode@@"_fnv1a: //CSectorDataNode
+				{
+					CSectorDataNode* sector_node = dynamic_cast<CSectorDataNode*>(node);
+					if (sector_node->m_pos_x == 712 || sector_node->m_pos_y == 712 || sector_node->m_pos_z == 712)
+					{
+						if (g->notifications.invalid_sync.log)
+							LOG(WARNING) << "Invalid sector position." << " From : " << sender->get_name();
+						if (g->notifications.invalid_sync.notify)
+							g_notification_service->push_warning(fmt::format("Invalid sector position from {}", sender->get_name()), "Invalid sector position.");
+						return true;
+					}
+					break;
+				}
+			}
```

Comment:
This is already become a lot of small checks, though it might be a hassle splitting each and every check into smaller files.

Maybe we could switch this over to an std::unordered_map where `typeid(*node).hash_code()` is the key and the value is a function which checks the data node.

Reply:
Nah